src/test/java/org/folio/rest/api/IterationJobRunnerTest.java

```diff
@@ -0,0 +1,134 @@
+package org.folio.rest.api;
+
+import static io.vertx.core.Future.succeededFuture;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.folio.okapi.common.XOkapiHeaders.TENANT;
+import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
+import static org.folio.rest.jaxrs.model.IterationJob.JobStatus.IDS_PUBLISHED;
+import static org.folio.rest.jaxrs.model.IterationJob.JobStatus.ID_PUBLISHING_CANCELLED;
+import static org.folio.rest.jaxrs.model.IterationJob.JobStatus.IN_PROGRESS;
+import static org.folio.rest.persist.PgUtil.postgresClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.UUID;
+
+import io.vertx.core.Context;
+import org.folio.persist.IterationJobRepository;
+import org.folio.rest.jaxrs.model.IterationJob;
+import org.folio.rest.jaxrs.model.IterationJobParams;
+import org.folio.rest.persist.PostgresClientFuturized;
+import org.folio.rest.support.kafka.FakeKafkaConsumer;
+import org.folio.rest.support.sql.TestRowStream;
+import org.folio.services.iteration.IterationJobRunner;
+import org.junit.Test;
+
+public class IterationJobRunnerTest extends TestBaseWithInventoryUtil {
+
+  // use usual instance topic for testing purposes, because it doesn't matter
+  // what the topic is for testing. this also prevents from adding changes to
+  // FakeKafkaConsumer
+  private static final String TEST_TOPIC = "inventory.instance";
+
+  private final IterationJobRepository repository = getRepository();
+
+
+  @Test
+  public void canIterateInstances() {
+    var numberOfRecords = 1100;
+    var rowStream = new TestRowStream(numberOfRecords);
+    var iterationJob = iterationJob();
+    var postgresClientFuturized = spy(getPostgresClientFuturized());
+
+    doReturn(succeededFuture(rowStream))
+      .when(postgresClientFuturized).selectStream(any(), anyString());
+
+    get(repository.save(iterationJob.getId(), iterationJob).toCompletionStage()
+      .toCompletableFuture());
```

Comment:
Extracting an overload of the `get` method that takes the return type of the `save` method and does the `.completionStage().toCompletableFuture()` could help reduce some of the noise from asynchronous operations on this line.

Reply:
Updated. Thanks